Doc/howto/clinic.rst

```diff
@@ -1891,3 +1891,82 @@ blocks embedded in Python files look slightly different.  They look like this:
     #[python start generated code]*/
     def foo(): pass
     #/*[python checksum:...]*/
+
+
+How to deprecate positional use of optional parameters
+------------------------------------------------------
+
+Argument Clinic provides syntax that makes it possible to generate code that
+deprecates positional use of optional parameters.
+For example, say we've got a module level function :py:func:`!foo.myfunc` that
+takes three :class:`int` parameters:
+optional parameters *a* and *b*, and keyword-only parameter *c*::
+
+   /*[clinic input]
+   module foo
+   myfunc
+       a: int
+       b: int
+       *
+       c: int
+   [clinic start generated output]*/
+
+We now want to make the *b* parameter keyword-only from Python 3.14 and onward,
+and since :py:func:`!myfunc` is a public API,
+we must follow :pep:`387` and issue a deprecation warning.
+We can do that using the ``* [from ...]``` syntax,
+by adding the line ``* [from 3.14]`` right above the *b* parameter::
+
+   /*[clinic input]
+   module foo
+   myfunc
+       a: int
+       * [from 3.14]
+       b: int
+       *
+       c: int
+   [clinic start generated output]*/
+
+Next, regenerate Argument Clinic code (``make clinic``),
+and add unit tests for the new behaviour.
+
+The generated code will now emit a :exc:`DeprecationWarning`
+when an the optional parameter *b* is used as a positional paremeter.
+C preprocessor directives are also generated for emitting
+compiler warnings if the ``* [from ...]`` line has not been removed
+from the Argument Clinic input when the deprecation period is over,
+which means when the beta phase of the specified Python version kicks in.
+
+Let's return to our example and assume Python 3.14 development has entered the
+beta phase, and we forgot all about updating the Argument Clinic code for
+:py:func:`!myfunc`.
+Luckily for us, compiler warnings are now generated:
+
+.. code-block:: none
+
+   In file included from Modules/foomodule.c:139:
+   Modules/clinic/foomodule.c.h:83:8: warning: Update 'b' in 'myfunc' in 'foomodule.c' to be keyword-only. [-W#warnings]
+    #    warning "Update 'b' in 'myfunc' in 'foomodule.c' to be keyword-only."
+         ^
+
+We now close the deprecation phase by making *b* keyword-only;
+replace the ``* [from ...]``` line above *b*
+with the ``*`` from the line above *c*::
+
+   /*[clinic input]
+   module foo
+   myfunc
+       a: int
+       *
+       b: int
+       c: int
+   [clinic start generated output]*/
+
+Finally, run ``make clinic`` to regenerate the Argument Clinic code,
+and update your unit tests to reflect the new behaviour.
```

Comment:
(*Grumbles about how `make clinic` doesn't work on Windows* -- but I'm guessing that's probably a broader issue across this piece of documentation, and if so, it shouldn't really be addressed in this PR :)